tensorflow/python/estimator/canned/boosted_trees.py

```diff
@@ -910,8 +938,97 @@ def _create_regression_head(label_dimension, weight_column=None):
   # pylint: enable=protected-access
 
 
+def _compute_feature_importance_for_tree(tree, num_features, normalize):
+  importances = np.zeros(num_features)
+
+  for node in tree.nodes:
+    node_type = node.WhichOneof('node')
+    if node_type == 'bucketized_split':
+      feature_id = node.bucketized_split.feature_id
+      importances[feature_id] += node.metadata.gain
+    elif node_type == 'leaf':
+      assert node.metadata.gain == 0
+    else:
+      raise ValueError('Unexpected split type %s', node_type)
+
+  if normalize:
+    normalizer = np.sum(importances)
+    if normalizer > 0.0:
+      # Avoid dividing by zero (e.g., when root is pure)
+      importances /= normalizer
+
+  return importances
+
+
+def compute_feature_importances(tree_ensemble,
+                                num_features,
+                                normalize=True):
+  tree_importances = [_compute_feature_importance_for_tree(tree,
+                                                           num_features,
+                                                           normalize)
+                      for tree in tree_ensemble.trees]
+  tree_importances = np.array(tree_importances)
+  tree_weights = np.array(tree_ensemble.tree_weights).reshape(-1, 1)
+  feature_importances = np.sum(tree_importances * tree_weights,
+                               axis=0) / np.sum(tree_weights)
+  if normalize:
+    normalizer = np.sum(feature_importances)
+    if normalizer > 0.0:
+      feature_importances /= normalizer
+
+  sorted_feature_idx = np.argsort(feature_importances)[::-1]
+  return sorted_feature_idx, feature_importances[sorted_feature_idx]
+
+
+class _BoostedTrees(estimator.Estimator):
+
+  def __init__(self, model_fn, model_dir, config, feature_columns):
+    super(_BoostedTrees, self).__init__(
+        model_fn=model_fn, model_dir=model_dir, config=config)
+
+    self._sorted_feature_columns = sorted(feature_columns, key=lambda tc: tc.name)
+
+  def compute_feature_importances(self, normalize=True):
+    tree_ensemble = self._read_tree_ensemble_from_checkpoint()
```

Comment:
May be lets do this
1) Add a method called experimental_explain_model(self, some params)
If it is experimental, it is allowed to change later. For example, we will add some other way of calculating feature importances, like not based on gain but just frequency of splitting etc
2) It will return a dict with {"feature_importances": ... } for now. Later we could extend it to return more things without having to change the interface

Reply:
Good idea. User can normalize the feature importance by themselves if necessary.

Reply:
I recommend 'experimental_feature_importances'. since it's more specific.